checker-util/src/main/java/org/checkerframework/checker/regex/util/RegexUtil.java

```diff
@@ -452,6 +452,9 @@ public static boolean noStringMatchesAnyRegex(
      * @param iterable an iterable
      * @return a list of the results of applying {@code f} to the elements of {@code iterable}
      */
+    @SuppressWarnings(
```

Comment:
@Ao-senXiong Can you try to understand why the behavior for the two test cases in the PR changed?
I tried to only clean-up the code, but these two tests changed behavior.
The changes do make sense to me, but they do come as a bit of a surprise.

Reply:
Hmm, also `:framework:checkInterning` and `:dataflow:checkResourceLeak` start to fail, so I must have goofed up somewhere...

Reply:
It would also be nice to minimize test cases for these two type systems, to detect these problems earlier.

Reply:
Working on this.

Reply:
@wmdietl Here is the minimal test case to reproduce interning checking failure in this branch:
```
public class Demo {

    public <T extends Dummy<?,?>> T test(){
        T t = getT();
        return t;
    }

    public <T extends Dummy<?, ?>> T getT(){
        return null;
    }

    public class Dummy<A,B>{

    }
}
```
```
java -jar checker/dist/checker.jar -processor org.checkerframework.checker.interning.InterningChecker Demo.java
Demo.java:5: error: [return.type.incompatible] incompatible types in return.
        return t;
               ^
  type of expression: T[ extends @UnknownInterned Demo.@UnknownInterned Dummy<?[ extends @UnknownInterned Object super @InternedDistinct Void], ?[ extends @UnknownInterned Object super @InternedDistinct Void]> super @UnknownInterned Void]
  method return type: T[ extends @UnknownInterned Demo.@UnknownInterned Dummy<?[ extends @UnknownInterned Object super @InternedDistinct Void], ?[ extends @UnknownInterned Object super @InternedDistinct Void]> super @InternedDistinct Void]
```

Reply:
A follow up here, looks like the type refinement works fine, however, the actual type of right hand side is changed.
In master:
```
InterningVisitor about to test whether actual is a subtype of expected (at Demo.java:4:15): actual tree = METHOD_INVOCATION getT()
     actual: TYPEVAR T[ extends @UnknownInterned Demo.@UnknownInterned Dummy<?[ extends @UnknownInterned Object super @InternedDistinct Void], ?[ extends @UnknownInterned Object super @InternedDistinct Void]> super @InternedDistinct Void]
   expected: TYPEVAR T[ extends @UnknownInterned Demo.@UnknownInterned Dummy<?[ extends @UnknownInterned Object super @InternedDistinct Void], ?[ extends @UnknownInterned Object super @InternedDistinct Void]> super @UnknownInterned Void]
```
In this branch
```
InterningVisitor about to test whether actual is a subtype of expected (at Demo.java:4:15): actual tree = METHOD_INVOCATION getT()
     actual: TYPEVAR T[ extends @UnknownInterned Demo.@UnknownInterned Dummy<?[ extends @UnknownInterned Object super @InternedDistinct Void], ?[ extends @UnknownInterned Object super @InternedDistinct Void]> super @UnknownInterned Void]
   expected: TYPEVAR T[ extends @UnknownInterned Demo.@UnknownInterned Dummy<?[ extends @UnknownInterned Object super @InternedDistinct Void], ?[ extends @UnknownInterned Object super @InternedDistinct Void]> super @UnknownInterned Void]
```

Reply:
This can also be confirmed by this test case works fine in this branch. I will debug and see why the right hand side of return type of method invocation is changed. 
```
public class Demo {

    public <T extends Dummy<?,?>> T test(){
        T t = null;
        return t;
    }

    public class Dummy<A,B>{

    }
}
```